src/pages/iou/steps/MoneyRequestConfirmPage.js

```diff
@@ -61,20 +61,30 @@ const defaultProps = {
 };
 
 const MoneyRequestConfirmPage = (props) => {
+    const prevMoneyRequestId = useRef(props.iou.id);
     const iouType = useRef(lodashGet(props.route, 'params.iouType', ''));
     const reportID = useRef(lodashGet(props.route, 'params.reportID', ''));
 
     useEffect(() => {
         const moneyRequestId = `${iouType.current}${reportID.current}`;
-        const shouldReset = props.iou.id !== moneyRequestId;
-        if (shouldReset) {
+        // ID in Onyx could change by initiating a new request in a separate browser tab
+        const isMoneyRequestIdChange = prevMoneyRequestId.current !== props.iou.id;
+        const isMoneyRequestIdMatch = props.iou.id === moneyRequestId;
```

Comment:
Why we are checking for both change and match?

Reply:
We can rewrite it like this to make it easier to read:
```
useEffect(() => {
    const moneyRequestId = `${iouType.current}${reportID.current}`;

    if (prevMoneyRequestId !== props.iou.id) {
        Navigation.goBack(ROUTES.getMoneyRequestRoute(iouType.current, reportID.current), true);
        return;
    }

    // Reset the money request Onyx if the ID in Onyx does not match the ID from params
    // and is not caused by an ID change in Onyx.
    const shouldReset = props.iou.id !== moneyRequestId;
    if (props.iou.id !== moneyRequestId) {
        IOU.resetMoneyRequestInfo(moneyRequestId);
    }

    if (_.isEmpty(props.iou.participants) || props.iou.amount === 0 || shouldReset) {
        Navigation.goBack(ROUTES.getMoneyRequestRoute(iouType.current, reportID.current), true);
    }
}, [props.iou.participants, props.iou.amount, props.iou.id]);
```

Reply:
yeah that rewrite looks better.